File: backend/app/models/repair.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Float, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base
import enum


class RepairStatus(str, enum.Enum):
    SUBMITTED = "submitted"
    ACCEPTED = "accepted"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    ACCEPTED_BY_TENANT = "accepted_by_tenant"


class RepairUrgency(str, enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    EMERGENCY = "emergency"


class RepairCostResponsibility(str, enum.Enum):
    TENANT = "tenant"
    LANDLORD = "landlord"
    NEGOTIATED = "negotiated"


class Repair(Base):
    __tablename__ = "repairs"

    id = Column(Integer, primary_key=True, index=True)
    property_id = Column(Integer, ForeignKey("properties.id"), nullable=False)
    tenant_id = Column(Integer, ForeignKey("tenants.id"))
    contract_id = Column(Integer, ForeignKey("contracts.id"))
    
    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    urgency = Column(String(20), default=RepairUrgency.MEDIUM.value)
    photo_urls = Column(Text, default="")
    
    status = Column(String(30), default=RepairStatus.SUBMITTED.value)
    
    assigned_to = Column(String(100))
    cost_estimate = Column(Float, default=0.0)
    actual_cost = Column(Float, default=0.0)
    cost_responsibility = Column(String(20))
    
    landlord_note = Column(Text, default="")
    tenant_note = Column(Text, default="")
    
    submitted_at = Column(DateTime, default=datetime.utcnow)
    accepted_at = Column(DateTime)
    started_at = Column(DateTime)
    completed_at = Column(DateTime)
    accepted_by_tenant_at = Column(DateTime)
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    property = relationship("Property", back_populates="repairs")

File: frontend/src/lib/api/repair.ts
import { api } from './client';

export interface Repair {
  id: number;
  property_id: number;
  tenant_id?: number;
  contract_id?: number;
  title: string;
  description: string;
  urgency: string;
  photo_urls: string;
  status: string;
  assigned_to?: string;
  cost_estimate: number;
  actual_cost: number;
  cost_responsibility?: string;
  landlord_note: string;
  tenant_note: string;
  submitted_at: string;
  accepted_at?: string;
  started_at?: string;
  completed_at?: string;
  accepted_by_tenant_at?: string;
  created_at: string;
  updated_at: string;
}

export interface RepairCreate {
  property_id: number;
  tenant_id?: number;
  contract_id?: number;
  title: string;
  description: string;
  urgency?: string;
  photo_urls?: string;
}

export const repairApi = {
  list: (params?: { status?: string; property_id?: number; urgency?: string }) => {
    const parts: string[] = [];
    if (params?.status) parts.push(`status=${params.status}`);
    if (params?.property_id) parts.push(`property_id=${params.property_id}`);
    if (params?.urgency) parts.push(`urgency=${params.urgency}`);
    const qs = parts.length ? `?${parts.join('&')}` : '';
    return api.get<Repair[]>(`/repairs${qs}`);
  },
  get: (id: number) => api.get<Repair>(`/repairs/${id}`),
  create: (data: RepairCreate) => api.post<Repair>('/repairs', data),
  update: (id: number, data: Partial<RepairCreate> & { status?: string }) =>
    api.put<Repair>(`/repairs/${id}`, data),
  accept: (id: number) => api.post<Repair>(`/repairs/${id}/accept`),
  start: (id: number) => api.post<Repair>(`/repairs/${id}/start`),
  complete: (id: number) => api.post<Repair>(`/repairs/${id}/complete`),
  acceptByTenant: (id: number) => api.post<Repair>(`/repairs/${id}/accept-by-tenant`),
  remove: (id: number) => api.del(`/repairs/${id}`),
};
